### PR TITLE
Support multiple filter values using comma separation instead of multiple input fields

### PR DESCRIPTION
**Pull request by attekei**
Created this as a starting point for addressing #4330. Our current implementation of supporting multiple fields involves an "Add another value" button and multiple inputs:

screen shot 2017-03-21 at 16 54 39

In this PR, that is replaced with values separated by commas and a small help text, as suggested by @mazameli:

screen shot 2017-03-21 at 16 47 14

The comma separation can be pretty trivially replaced with newlines if that seems to be a better long-term approach.

### CODE REVIEW

**Comment by attekei** [comment]
Now the input field grows with content, max. 8 lines. Newlines are automatically stripped in case that user tries to paste some text containing newlines (normally pressing return commits the field value).

**Review by mazameli — CHANGES_REQUESTED** [review]
This is looking great other than the problem I mentioned with how we're not yet trimming leading or trailing spaces (so `value_1,value_2` currently works, but `value_1, value_2` does not). 

I also left a question in slack about whether or not we'd want to add this capability to other filter types beyond just `is` and `is not`, but that would be an enhancement.

File: frontend/src/metabase/css/core/text.css [hunk 1]
```diff
@@ -90,6 +90,10 @@
   line-height: 1.5em;
 }
 
+.text-small {
```
Comment (tlrobinson) [inline frontend/src/metabase/css/core/text.css]: Paging @kdoh

File: frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx [hunk 2]
```diff
@@ -29,63 +30,47 @@ export default class TextPicker extends Component<*, Props, *> {
...
-                                </span>
-                            : null }
-                        </li>
-                    )}
-                </ul>
+                <div className="FilterInput px1 pt1 relative">
+                    <AutosizeTextarea
+                        className={cx("input block full border-purple", { "border-error": hasInvalidValues })}
+                        type="text"
+                        value={values.join(',')}
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx]: likewise, `.join(", ")`

File: frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx [hunk 3]
```diff
@@ -29,63 +30,47 @@ export default class TextPicker extends Component<*, Props, *> {
...
-    }
-
-    setValue(index: number, value: string|null) {
-        let values = this.props.values.slice();
-        values[index] = value;
-        this.props.onValuesChange(values);
+    setValue(fieldString: ?string) {
+        if (fieldString != null && fieldString !== "") {
+            const newLineRegex = /\r?\n|\r/g;
+            const newValues = fieldString.replace(newLineRegex,'').split(',').map((value: string|null) => value);
```
Comment (tlrobinson) [inline frontend/src/metabase/query_builder/components/filters/pickers/TextPicker.jsx]: Probably want to do `.split(/,\s*/g)` so `foo, bar` doesn't become `["foo", " bar"]` (note the space before "bar")

**Review by tlrobinson — CHANGES_REQUESTED** [review]

**Comment by attekei** [comment]
Updated behavior:
- While editing the text, only newlines are filtered on-the-fly to offer a predictable copy-pasting experience
- Extra commas and whitespaces are tolerated but removed during commit
- Empty values are removed during commit

**Review by mazameli — APPROVED** [review]

**Review by tlrobinson — APPROVED** [review]

**Pull request merged**